Roihu updates to Docs CSC (#2597)
CSCfi/csc-user-guide: docs/support/tutorials/roihu.md

     Roihu web interface will only be vailable after General Availability. Please connect via SSH during the pilot phase.
     

The simplest way to connect to Roihu is to use the web interface.

1. Go to [www.roihu.csc.fi](https://www.roihu.csc.fi).
2. Log in using your Haka, Virtu or CSC user account.
   [Multi-factor authentication (MFA)](../../accounts/mfa.md) is required.

## Migrating research data

If you need to transfer data from Puhti or Mahti to Roihu, we require
that you:

1. Review your data carefully – **only move what you really need**
2. Check your available disk space on Roihu (for example, using the `csc-workspaces` command)
3. Transfer data **directly** from Puhti or Mahti to Roihu.

Note that previous extended disk quotas on Puhti or Mahti will not be automatically moved to Roihu.
Quota extensions on Roihu must be separately applied for and properly motivated.

**[Read the detailed instructions in the Roihu data migration guide](roihu-data.md).**

## Installing software

Before installing anything:

1. Check if the software is already available:
     - [List of pre-installed applications](../../apps/by_availability.md#roihu)

>      - `module spider <module name>`

Reviewer: ```suggestion
Before installing anything check if the software is already available:
     - [List of pre-installed applications](../../apps/by_availability.md#roihu)
     - `module spider <software name>`
```

Numbered list with only one item looks weird.

If not available, choose one of the following approaches depending on your needs:

### Compiling C/C++/Fortran code

HPC software written using programming languages such as C, C++ or Fortran need to be compiled before installing.  
For instructions on the available compilers and preferred options, see the instructions for compiling software on:

- [Compiling on Roihu-CPU](../../computing/compiling-roihu.md#compiling-on-roihu-cpu)
- [Compiling on Roihu-GPU](../../computing/compiling-roihu.md#compiling-on-roihu-gpu)

### Containers

Roihu supports Apptainer/Singularity containers for container installations. 
In most cases, ready-made Docker containers can be easily converted into an Apptainer image.
Another option is to build your own container from scratch. 
You can build containers on top of Roihu base containers which have the same software stack as is available via the module system natively.

> Base container are built on top of Rockylinux 9.

Reviewer: ```suggestion
Base container are built on top of Rocky Linux 9.
```

=== "Roihu CPU base container (~4 GB)"
    ```sh title="container.def"
    Bootstrap: docker
    From: satama.csc.fi/r_installation_spack/core-cpu-gcc-15.2.0:v2026_03

    %post
        # Activate module environment and load default modules.
        . /opt/activate.sh
        # Build your application here:

    %runscript
        . /opt/activate.sh
        exec "$@"
    ```

>     When building the containers, set you cache directory to temporary directory to avoid filling you home directory quota.

Reviewer: ```suggestion
    When building containers, set the Apptainer cache directory to `$TMPDIR` to avoid filling your home directory quota.
```

    ```bash
    export APPTAINER_CACHEDIR=$TMPDIR
    apptainer build --fakeroot container.sif container.def
    ```

    Now, you can run commands inside the container with clean environment and environment active as follows:

    ```bash
    apptainer run --cleanenv run mycmd
    ```

=== "Roihu GPU base container (~16 GB)"
    ```sh title="container.def"
    Bootstrap: docker
    From: satama.csc.fi/r_installation_spack/core-gpu-gcc-14.3.0-cuda-12.9.1

    %post
        # Activate module environment and load default modules.
        . /opt/activate.sh
        # Build your application here:

    %runscript
        . /opt/activate.sh
        exec "$@"
    ```

    When building the containers, set you cache directory to temporary directory to avoid filling you home directory quota.

    ```bash
    export APPTAINER_CACHEDIR=$TMPDIR
    apptainer build --fakeroot container.sif container.def
    ```

    Now, you can run commands inside the container with clean environment and environment active as follows:

    ```bash
    apptainer run --cleanenv run mycmd
    ```

> More details on working with containers in CSC's computing environment can be found from the links below:

Reviewer: Should we also mention the machine learning base and PyTorch base containers here as they have a slightly different approach?

Reviewer 2: To discuss with @jaantollander

- [Overview of containers](../../computing/containers/overview.md)
- [Running containers](../../computing/containers/overview.md#running-containers)
- [Creating containers](../../computing/containers/overview.md#building-container-images)
- [Tykky container wrapper](../../computing/containers/tykky.md)

### Python/R environments

Best practice guidelines on installing your own Python and R packages can be found in the Python, R and Tykky container wrapper pages below.

- [Installing Python packages and environments](../tutorials/python-usage-guide.md)
- [Containerizing Conda and pip environments with Tykky](../../computing/containers/tykky.md)
- [R package installations](../../apps/r-env.md#r-package-installations)

## Running your first job

Roihu uses Slurm, similarly to Puhti and Mahti.

Basic workflow:

1. Create a job script where you
    * Define the resources for your job (time, memory, cores)
    * Load the required modules
    * Launch your executable
2. Submit your batch job into the queuing system
3. Wait for the job to finish, and look for its output

See the relevant documentation below for detailed information:

> 1. [Available batch job partitions](../../computing/running/batch-job-partitions.md)

Reviewer: The page linked to here does not actually have any Roihu partitions!

Reviewer 2: Updated Roihu Slurm docs currently under:
https://github.com/CSCfi/csc-user-guide/pull/2939 , and
https://github.com/CSCfi/csc-user-guide/pull/2894

Should be merged ASAP

Reviewer 2: Updated here: https://github.com/CSCfi/csc-user-guide/pull/2597/commits/631ad78d03d1c450610760a076e8e2731a91f397

2. [Creating a batch job script](../../computing/running/batch-job-partitions.md)
3. [Submit a batch job](../../computing/running/submitting-jobs.md)
4. [Performance checklist](../../computing/running/performance-checklist.md)

For common Slurm error messages, see our FAQ on [Why does my batch job fail?](../faq/why-does-my-batch-job-fail.md).

### Known issues (pilot phase)

During the pilot phase, you may encounter multiple warnings or errors related to *Argos* in your Slurm job output, for example:

```
error: argos:slurm_spank_task_init: get_env_var: cannot get SLURM_ARGOS_SPANK_OPT from job(22474) environment (No such environment variable)
```

These messages are **harmless** and do not affect your job execution.
Your job will continue normally with Argos disabled.

If your job completes successfully, you can safely ignore these messages.

To suppress most of the Argos related warnings and errors, you can pass the `--argos=no` flag option to srun in the following manner:

```bash
#!/bin/bash
#SBATCH --account=<project_id>
#SBATCH --partition=test
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --time=DD:HH:MM

> srun --argos=no <your-executable>

Reviewer: Can't you also have it as an SBATCH option?

```
#SBATCH --argos=no
```
